### PR TITLE
Move Docker client to Docker infrastructure packaging

### LINKED ISSUE #6404: SPI: Move docker infrastructure related code from docker-plugin to docker infrastructure
We have code that is placed in module plugin-docker but it is not a plugin at all comparing to what we have in other plugins. It is rather infrastructure related code. So we should move this code into related infrastructure modules.

**Reproduction Steps:**

**OS and version:**    

**Diagnostics:**

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Move Docker client to Docker infrastructure packaging.
Also,  remove OpenShiftConnector as it is replaced with OpenShift
infrastructure.

### What issues does this PR fix or reference?
Fixes #6404

### CODE REVIEW

**Review by riuvshin — APPROVED** [review]

File: infrastructures/docker/pom.xml [hunk 1]
```diff
@@ -15,185 +15,31 @@
...
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <configuration>
-                    <excludes>
-                        <exclude>**/integration/**</exclude>
-                    </excludes>
-                </configuration>
-            </plugin>
-        </plugins>
+        <pluginManagement>
```
Comment (benoitf) [inline infrastructures/docker/pom.xml]: do we need this pluginManagement section here ?

**Review by skabashnyuk — APPROVED** [review]

File: infrastructures/docker/pom.xml (hunk 1 above)
Comment (garagatyi) [reply infrastructures/docker/pom.xml]: removed

File: infrastructures/pom.xml [hunk 2]
```diff
@@ -19,7 +19,9 @@
         <version>5.20.0-SNAPSHOT</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
+    <groupId>org.eclipse.che.infrastructure</groupId>
     <artifactId>che-infrastructures-parent</artifactId>
+    <version>5.20.0-SNAPSHOT</version>
```
Comment (sleshchenko) [inline infrastructures/pom.xml]: I think you can remove version here

File: infrastructures/docker/infrastructure/pom.xml [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <exclude>**/integration/**</exclude>
```
Comment (sleshchenko) [inline infrastructures/docker/infrastructure/pom.xml]: I don't see `integration` package. Please check that it is needed here

**Review by sleshchenko — APPROVED** [review]

File: infrastructures/pom.xml (hunk 2 above)
Comment (garagatyi) [reply infrastructures/pom.xml]: ok

File: infrastructures/docker/infrastructure/pom.xml (hunk 3 above)
Comment (garagatyi) [reply infrastructures/docker/infrastructure/pom.xml]: Nice catch!

**Review by benoitf — APPROVED** [review]
good bye OpenShift Connector 😭

**Comment by garagatyi** [comment]
@benoitf You can still contribute to the new OpenShift infra instead and I hope it will be easier and more straightforward than what we have in master

**Comment by garagatyi** [comment]
@l0rd can you take a look on this PR?

**Review by l0rd — APPROVED** [review]
👍

**Pull request merged**